### PR TITLE
feat: AuthHandler performing basic GNAP AS logic

### PR DESCRIPTION
**Pull request by Moopli**
Signed-off-by: Filip Burlacu <filip.burlacu@securekey.com>

### CODE REVIEW

File: cmd/auth-rest/startcmd/start.go [hunk 1]
```diff
@@ -440,6 +443,11 @@ func startAuthService(parameters *authRestParameters, srv server) error {
 		router.HandleFunc(handler.Path(), handler.Handle()).Methods(handler.Method())
 	}
 
+	interact, err := redirect.New()
```
Comment (rolsonquadras) [inline cmd/auth-rest/startcmd/start.go:446]: We should add a TODO as AS need to support different flows.

**Review by rolsonquadras — APPROVED** [review]

File: cmd/auth-rest/startcmd/start.go (hunk 1 above)
Comment (Moopli) [reply cmd/auth-rest/startcmd/start.go:447]: TODO added

File: pkg/gnap/authhandler/auth_handler.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+	reqVerifier api.Verifier,
+) (*gnap.AuthResponse, error) {
+	var (
+		s   *session.Session
+		err error
+	)
+
+	if req.Client == nil {
+		// client can never be omitted entirely
+		return nil, fmt.Errorf("missing client")
```
Comment (Baha-sk) [inline pkg/gnap/authhandler/auth_handler.go:84]:
```suggestion
		return nil, errors.New("missing client")
```

and of course update the import

Comment (Moopli) [reply pkg/gnap/authhandler/auth_handler.go]: done

**Review by Baha-sk — APPROVED** [review]

**Pull request merged**